### PR TITLE
Partial workflow execution by steps

### PR DESCRIPTION
**Pull request by denisri**
* Introduce pipeline steps: groups of nodes which execution may be disabled in a workflow. Steps are defined in a pipeline, as a new trait: pipeline.pipeline_steps
* Workflows have been improved, they are now ordered in a logical way.
* GUI allows to use the steps enabling/disabling in PipelineDeveloperView and from a controller.
* More functions are also available to test a pipeline parameters before running: detect missing inputs, existing outputs, handle temporary files... and have a GUI to display problems.
* Older functions in pipeline_tools have been removed from the GUI. They are still here but deprecated, we may remove them soon.

### CODE REVIEW

File: capsul/pipeline/pipeline.py [hunk 1]
```diff
@@ -1380,3 +1454,118 @@ def compare_dict(ref_dict, other_dict):
...
+    def add_pipeline_step(self, step_name, nodes):
+        '''Add a step definiton to the pipeline (see also define_steps).
+
+        Steps are groups of pipeline nodes, which may be disabled at runtime.
+        They are normally defined in a logical order regarding the workflow
+        streams. They are different from pipelines in that steps are purely
+        virtual groups, they do not have parameters.
+
+        Disabling a step acts differently as the pipeline node activation: other
+        nodes are not inativated according to their dependencies. Instead,
```
Comment (sapetnioc) [inline capsul/pipeline/pipeline.py]: Typo inativated -> inactivated

File: capsul/qt_gui/widgets/full_pipeline_widgets.py [hunk 2]
```diff
@@ -1349,20 +1385,70 @@ def openPopupMenu(self, node_name, process):
...
-
-        disable_done_action = menu.addAction('Disable nodes with existing outputs')
-        disable_done_action.triggered.connect(self.disable_done_outputs)
-
-        reactivate_pipeline_action = menu.addAction('Reactivate disabed pipeline nodes')
-        reactivate_pipeline_action.triggered.connect(self.reactivate_pipeline)
-
-        reactivate_node_action = menu.addAction('Reactivate disabed pipeline node')
-        reactivate_node_action.triggered.connect(self.reactivate_node)
+        #disable_down_action = menu.addAction('Disable for downhill processing')
```
Comment (sapetnioc) [inline capsul/qt_gui/widgets/full_pipeline_widgets.py]: Is it necessary to keep these commented lines ?

Comment (sapetnioc) [inline capsul/qt_gui/widgets/full_pipeline_widgets.py]: Same question for other commented lines below.

File: capsul/qt_gui/widgets/pipeline_file_warning_widget.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+# System import
+from soma.qt_gui.qt_backend import QtCore, QtGui
+try:
+    import traits.api as traits
+except ImportError:
+    import enthought.traits.api as traits
+
+
+class PipelineFileWarningWidget(QtGui.QSplitter):
+
```
Comment (sapetnioc) [inline capsul/qt_gui/widgets/pipeline_file_warning_widget.py]: A bit of docstring to give some hint of the usage of this class in the GUI would be a plus.

**Comment by denisri** [comment]
OK I have fixed the issues.

**Pull request merged**